backfill/shared/dart_dink.go

```diff
@@ -21,11 +22,31 @@ var (
 type DartDink struct {
 	Dart     string
 	Dink     string
-	HeaderID int
-	UrnID    int
+	HeaderID int64
+	UrnID    int64
 }
 
-func FetchAndPersistDartDinkDiffs(dartDink DartDink, eventsRepository repository.EventsRepository, storageRepository repository.StorageRepository, blockChain core.BlockChain) error {
+type DartDinkRetriever interface {
+	RetrieveDartDinkDiffs(dartDink DartDink) error
+}
+
+type dartDinkRetriever struct {
+	blockChain        core.BlockChain
+	eventsRepository  repository.EventsRepository
+	headerRepository  datastore.HeaderRepository
+	storageRepository repository.StorageRepository
+}
+
+func NewDartDinkRetriever(blockChain core.BlockChain, eventsRepository repository.EventsRepository, headerRepository datastore.HeaderRepository, storageRepository repository.StorageRepository) DartDinkRetriever {
+	return dartDinkRetriever{
+		blockChain:        blockChain,
+		eventsRepository:  eventsRepository,
+		headerRepository:  headerRepository,
+		storageRepository: storageRepository,
+	}
+}
+
+func (retriever dartDinkRetriever) RetrieveDartDinkDiffs(dartDink DartDink) error {
```

Comment:
This is kind of nitpicky, but `RetrieveDartDinkDiff` seems like the method will be returning DartDinkDiffs to me, instead of the inserting them. But I'm not sure how else to name this besides going back to `FetchAndPersist...`. Definitely not merge blocking though!

Reply:
Yeah, this was a major struggle for me. I'm honestly just not sure how to name this, but `DartDinkFetchAndPersister` is where I drew the line 😅